### PR TITLE
wiki controller specs done

### PR DESCRIPTION
**Pull request by AbdullahAlger**
I think the specs are done. I did take one test out, which said that a standard user can update a wiki. The only reason I took it it was that this is tested in the model spec. Also, a standard user never sees the private option. The only issue I see here is that if some hacking went on, then, yes, a standard user would could have private wikis, but they'd never see them.

### CODE REVIEW

File: spec/controllers/wikis_controller_spec.rb [hunk 1]
```diff
@@ -63,8 +62,9 @@
       sign_in(user)
       params = {wiki: {title: "", body: "Angular testing causes brain malfunctions, brain malfunctions"}}
       post :create, params
+      wiki = Wiki.any?
```
Comment (eliotsykes) [inline spec/controllers/wikis_controller_spec.rb]: `wiki` is a misleading name for this variable - `Wiki.any?` returns a boolean, the name suggests it'd be a `Wiki` object. To workaround this, lines 65 and 66 could be combined into one line without using a local variable.

File: spec/controllers/wikis_controller_spec.rb [hunk 2]
```diff
@@ -73,8 +73,9 @@
       sign_in(user)
       params = {wiki: {title: "Angular is fun, except testing it", body: ""}}
       post :create, params
+      wiki = Wiki.any?
+      expect(wiki).to eq false
```
Comment (eliotsykes) [inline spec/controllers/wikis_controller_spec.rb]: See comment above

File: spec/controllers/wikis_controller_spec.rb [hunk 3]
```diff
@@ -107,20 +108,17 @@
...
       sign_in(user)
-      puts user.role
-      wiki = create(:wiki, user: user)
-      params = {wiki: {title: "new title", body: "new bodynew bodynew bodynew bodynew body", private: "1"}}
+      wiki = create(:wiki, user: user, private: false)
+      params = {wiki: {private: "1"}, id: wiki.slug}
 
-      post :create, params
+      patch :update, params
+      wiki.reload
```
Comment (eliotsykes) [inline spec/controllers/wikis_controller_spec.rb]: Good use of `.reload`

**Comment by AbdullahAlger** [comment]
That's your suggestion.

File: spec/controllers/wikis_controller_spec.rb (hunk 3 above)
Comment (AbdullahAlger) [inline spec/controllers/wikis_controller_spec.rb]: That's your suggestion

**Comment by eliotsykes** [comment]
> The only issue I see here is that if some hacking went on, then, yes, a standard user would could have private wikis, but they'd never see them. 

Admittedly low risk of hacking on this app, however it'd be good to get the practice in how to test this kind of thing that's hard to test fully at the model and feature spec levels. Let me know if you'd like to see what the spec would look like.

**Comment by AbdullahAlger** [comment]
Yep. I'd like to see it

**Comment by eliotsykes** [comment]
Added in #43

File: spec/controllers/wikis_controller_spec.rb (hunk 1 above)
Comment (AbdullahAlger) [inline spec/controllers/wikis_controller_spec.rb]: I get your point.

**Pull request merged**